scripts/src/mock_client.ts

```diff
@@ -4,50 +4,78 @@ import { Offer, OfferList } from "./services/offers";
 import { OpenOrder, Order, OrderList } from "./services/orders";
 import { Poll } from "./services/offer_contents";
 import { delay } from "./utils";
+import { Application } from "./models/applications";
+import { ApiError } from "./middleware";
 
 const BASE = "http://localhost:3000";
 
 class Client {
 
 	public token = "";
 
-	public async register(appId: string, userId: string, walletAddress: string) {
-		const res = await axios.default.post(BASE + "/v1/users", {
+	public async register(appId: string, apiKey: string, userId: string, walletAddress: string) {
+		const res = await this._post("/v1/users", {
 			sign_in_type: "whitelist",
 			user_id: userId,
 			device_id: "my_device",
 			app_id: appId,
+			api_key: Application.KIK_API_KEY,
 			public_address: walletAddress,
-		}, this.getConfig());
+		});
 
 		this.token = res.data.token;
 	}
 
+	public async activate() {
+		const res = await this._post("/v1/users/me/activate");
+		this.token = res.data.token;
+	}
+
 	public async getOffers(): Promise<OfferList> {
-		const res = await axios.default.get(BASE + "/v1/offers", this.getConfig());
+		const res = await this._get("/v1/offers");
 		return res.data as OfferList;
 	}
 
 	public async createOrder(offerId: string): Promise<OpenOrder> {
-		const res = await axios.default.post(BASE + `/v1/offers/${offerId}/orders`, {}, this.getConfig());
+		const res = await this._post(`/v1/offers/${offerId}/orders`);
 		return res.data as OpenOrder;
 	}
 
 	public async submitOrder(orderId: string, content: string): Promise<Order> {
-		const res = await axios.default.post(BASE + `/v1/orders/${orderId}`, { content }, this.getConfig());
+		const res = await this._post(`/v1/orders/${orderId}`, { content });
 		return res.data as Order;
 	}
 
 	public async getOrder(orderId: string): Promise<Order> {
-		const res = await axios.default.get(BASE + `/v1/orders/${orderId}`, this.getConfig());
+		const res = await this._get(`/v1/orders/${orderId}`);
 		return res.data as Order;
 	}
 
 	public async getOrders(): Promise<OrderList> {
-		const res = await axios.default.get(BASE + "/v1/orders", this.getConfig());
+		const res = await this._get("/v1/orders");
 		return res.data as OrderList;
 	}
 
+	private async _get(url: string): Promise<any> {
+		try {
+			return await axios.default.get(BASE + url, this.getConfig());
+		} catch (error) {
+			const ex: axios.AxiosError = error;
```

Comment:
this happens at least twice, maybe turn it into a function:
```
function handleAxiosError(ex: axios.AxiosError): never {
	const apiError: ApiError = ex.response.data;
	throw Error(`server error ${ex.response.status}(${apiError.status}): ${apiError.error}`);
}
```
(it will also give us a chance the to use the `never` type!)